lib/pub_sub/subscriber.rb

```diff
@@ -5,23 +5,55 @@ def self.subscribe
     end
 
     def subscribe
-      PubSub.config.subscriptions.keys.each do |service_identifier|
+      topics = PubSub.config.subscriptions.keys.map do |service_identifier|
         subscribe_to_service(service_identifier)
       end
+      set_queue_policy(topics.map(&:arn))
     end
 
+    private
+
     def subscribe_to_service(service_identifier)
       topic = Aws::SNS::Topic.new(sns.create_topic(name: service_identifier).topic_arn)
-      queue_data = Aws::SQS::Client.new.get_queue_attributes(queue_url: Queue.new.queue_url, attribute_names: ["QueueArn"])
-      queue_arn = queue_data.attributes["QueueArn"]
       topic.subscribe(endpoint: queue_arn, protocol: 'sqs')
       PubSub.logger.info "Subscribed #{queue_arn} to #{topic.arn}"
+      topic
     end
 
-    private
+    def set_queue_policy(topic_arns)
```

Comment:
I believe this overwrites the current policy with the new one. In order to add multiple policies, you'd have to fetch the current policy, append the new one, then write it back.  Also, this passes multiple topic arns, which allows multiple topics to post to this queue from a single policy statement.  There are a few subscriptions that currently exist which have a separate statement for each topic arn.  These should probably be cleaned up to use an array of topics